compliance/src/main/java/bio/terra/pearl/compliance/SyncVantaUsers.java

```diff
@@ -0,0 +1,296 @@
+
+            if (response != null) {
+                if (response.getResults() != null) {
+                    elements.get().addAll(response.getResults().getData());
+                }
+            } else {
+                break;
+            }
+        }
+        return elements.get();
+    }
+
+    private WebClient getWebClientForIntegration(String accessToken, String integrationId, String resourceKind) {
+        WebClient.Builder webClientBuilder = WebClient.builder().defaultHeaders(h -> {
+            h.setBearerAuth(accessToken);
+            h.setContentType(MediaType.APPLICATION_JSON);
+        });
+
+        return webClientBuilder.baseUrl(vantaBasePath + "v1/integrations/" + integrationId + "/resource-kinds/" + resourceKind + "/resources").build();
+    }
+
+    public String syncVantaAccounts() {
+        StringBuilder summary = new StringBuilder();
+        WebClient wc = WebClient.builder().baseUrl(vantaBasePath + "oauth/token").build();
+        AccessToken vantaToken= wc.post().bodyValue(new VantaCredentials("client_credentials", "vanta-api.all:read vanta-api.all:write", vantaClientId, vantaSecret))
+                .header("Content-Type", "application/json").retrieve().bodyToMono(AccessToken.class).block();
+        WebClient.builder().defaultHeaders(h -> {
+            h.setBearerAuth(vantaToken.getAccess_token());
+            h.setContentType(MediaType.APPLICATION_JSON);
+        });
+
+        // go through each integration and find scope changes that need to be made.
+        int resourceBatchSize = 50;
+        Collection<VantaObject> vantaObjects = new HashSet<>();
+        List<VantaIntegration> integrationsToSync = new ArrayList<>();
+        integrationsToSync.add(new VantaIntegration("jira", "JiraAccount", new ParameterizedTypeReference<VantaResultsResponse<JiraAccount>>() {}, Set.of("669a8422865ac5731466a323", "66c629d278231843aec354ff", "669a8431865ac5731469ed44",
+                "669a842e865ac5731469663b", "669a8427865ac5731467c5d0", "669a8431865ac5731469e994")));
+        integrationsToSync.add(new VantaIntegration("slack", "SlackAccount", new ParameterizedTypeReference<VantaResultsResponse<SlackUser>>() {}, Set.of("66a14ed517f6ad4ce8ea8e7f","66a14ee917f6ad4ce8ef12ff","66c74e0878231843aea7b330")));
+        integrationsToSync.add(new VantaIntegration("github", "GithubAccount", new ParameterizedTypeReference<VantaResultsResponse<GithubAccount>>() {},
+                Set.of("66831806bb4f7b57d3b1b260", "66831804bb4f7b57d3b15f0c", "66831809bb4f7b57d3b24ecf", "66831803bb4f7b57d3b122b7", "66831802bb4f7b57d3b0c4d3",
+                        "6683180abb4f7b57d3b2ac3c", "66831809bb4f7b57d3b23dc1", "668de80159091c1ee96b8e4a", "66831808bb4f7b57d3b2144b")));
+        integrationsToSync.add(new VantaIntegration("workday", "WorkdayHrUser", new ParameterizedTypeReference<VantaResultsResponse<WorkdayAccount>>() {}, Collections.emptySet()));
+
+        for (VantaIntegration vantaIntegration : integrationsToSync) {
```

Comment:
can you pull the contents of this for loop into a separate method just for clarity?